packages/client/src/v2-events/components/forms/FormFieldGenerator.tsx

```diff
@@ -628,6 +654,22 @@ class FormSectionComponent extends React.Component<AllProps> {
             >
               <Field name={field.id}>
                 {(formikFieldProps: FieldProps<any>) => {
+                  const defaultValue =
+                    'defaultValue' in field &&
+                    typeof field.defaultValue != 'object' && // currently the default values do not support the 'complex' types, only strings, numbers and boolean
+                    field.defaultValue
+
+                  // Logic for using default value if defined
+                  useEffect(() => {
+                    const shouldUseDefaultValue =
+                      defaultValue && formikFieldProps.field.value === undefined
+
+                    if (shouldUseDefaultValue) {
+                      formikFieldProps.field.value = defaultValue
+                      this.setFieldValuesWithDependency(field.id, defaultValue)
+                    }
+                  }, [defaultValue, field.id, formikFieldProps.field])
+
```

Comment:
I think this should be further up in the logic.

```ts
  return (
    <Formik<ActionFormData>
      enableReinitialize={true}
      initialValues={initialValues}
```

I think so that the default values get merged in already here where we pass in the initialValues. This is how it works also in the [old implementation](https://github.com/opencrvs/opencrvs-core/blob/c6510248ea290eed7e81d2252832c9a60020ad0d/packages/client/src/components/form/FormFieldGenerator.tsx#L1343-L1354)